cobigen-setup/index.asciidoc

```diff
@@ -0,0 +1,89 @@
+= A setup tutorial for devonfw Code-based incremental Generator(i.e. Cobigen)
+====
+CobiGen is a generic incremental code generator. It allows you to build Java CRUD application based on the devonfw architecture including all software layers. You can generate all necessary classes and services (DAOs, Transfer Objects, simple CRUD use cases with REST services) of the application from one single entity class.
+
+Here in this tutorial we will learn how we can set up the CobiGen environment in our system along with eclipse and maven integration. 
+
+Cobigen provides eclipse and maven integration. There is command line interface(CLI) through which we can proceed with integration.
+
+More information about CobiGen on https://devonfw.com/website/pages/docs/master-cobigen.asciidoc.html
+====
+
+[step]
+--
+restoreDevonfwIde(["java","mvn"], "2020.08.001")
+--
+
+====
+# Prerequisites
+Before we start, the installation of devonfw ide is already done for you.
+Now we will proceed with CobiGen installation.
+[step]
+--
+installCobiGen()
+--
+
+In order to execute the commands we will have to initialize the CobiGen CLI first.
+
+====
+
+[step]
+--
+nextKatacodaStep("CLI Execution steps", [{ "file": "files/cliExecutionCommands.md" }])
+--
+
+====
+Below are the commands which we can use on CobiGen CLI.
+Using the following command and option you will be able to customize your generation as follows
+
+[step]
+--
+nextKatacodaStep("Commands and Options", [{ "file": "files/commandsAndOptions.md" }])
+--
+
+To get more information about Troubleshooting in CLI visit on https://devonfw.com/website/pages/docs/master-cobigen.asciidoc_cobigen-cli.html#howto_cobigen-cli-generation.asciidoc_troubleshooting
+
+Now in the next step we will see how to integrate CobiGen in eclipse.
+====
+
+====
+In this step we will learn about how we can integrate CobiGen in eclipse.
+
+Remark: CobiGen is preinstalled in the devonfw/devon-ide.
+
+1.1  Preconditions
+    - Eclipse 4.x
+
+    - Java 7 Runtime (for starting eclipse with CobiGen). This is independent from the target version of your developed code
+
+1.2. Installation steps
+    1. Open the eclipse installation dialog
+    menu bar → Help → Install new Software
+[step]
+--
+nextKatacodaStep("Installation", [{"image": "images/01-install-new-software.png"}])
+-- 
+
+    Here we can select the updates.
+    2. Open CobiGen’s update site

```

Comment:
This text belongs to the picture on the next step. Either move the text to the next step or display the picture in this step